packages/codemirror/src/editor.ts

```diff
@@ -742,7 +742,19 @@ export class CodeMirrorEditor implements CodeEditor.IEditor {
       // Only render selections if the start is not equal to the end.
       // In that case, we don't need to render the cursor.
       if (!JSONExt.deepEqual(selection.start, selection.end)) {
-        const { anchor, head } = this._toCodeMirrorSelection(selection);
+        // Selections only appear to render correctly if the anchor
+        // is before the head in the document. That is, reverse selections
+        // do not appear as intended.
+        let forward: boolean =
```

Comment:
Why did you take this logic out of the function below, and hardcode it here? The function below is called other places - was it not needed in the other places?

Reply:
This logic prevents reverse cursors from being set on the document, which I needed to ensure that the search and replace proceed correctly.  @ian-r-rose originally added this logic in #2139 as part of early RTC work.  He said it should be safe to move that logic here, since that RTC collaborator cursor code isn't really exercised any more.